src/components/button/Button.js

```diff
@@ -0,0 +1,44 @@
+import './styles.css';
+
+class Button {
+  constructor({ label = '', onClick = undefined, type = 'default', imageSrc = null, id, width = 'auto' }) {
+    this.is = id;
+    this.label = label;
+    this.onClick = onClick;
+    this.type = type;
+    this.imageSrc = imageSrc;
+    this.width = width;
+  }
+
+  render() {
+    const btn = document.createElement(`button`);
+    const img = document.createElement(`img`);
+    img.src = this.imageSrc;
+    btn.appendChild(img);
+    btn.innerText = this.label;
+
```

Comment:
btn always has a class button so you can write
btn.classList.add('button')

before all those 'ifs'